src/librustc_errors/lib.rs

```diff
@@ -103,6 +104,7 @@ pub trait CodeMapper {
     fn span_to_filename(&self, sp: Span) -> FileName;
     fn merge_spans(&self, sp_lhs: Span, sp_rhs: Span) -> Option<Span>;
     fn call_span_if_macro(&self, sp: Span) -> Span;
+    fn load_source_for_filemap(&self, file: FileName) -> bool;
```

Comment:
I find it a bit ridiculous that `CodeMap` has to be abstracted away because `rustc_errors` doesn't contain it *and* `FileMap` can't even see this trait.